tools/scripts/autorun_generateSoundMetadata.rb

```diff
@@ -0,0 +1,123 @@
+# example command = bundle exec ./parse_new_sound_aliases_csv.rb ~/Downloads/Animals ./sound_metadata.csv
+require 'csv'
+require 'fileutils'
+
+class NewSoundAliasesMetadataBuilder
+  def initialize(csv)
+    @csv = CSV.parse(File.read(csv), headers: true)
+  end
+
+  def fix_corrupted_filename(name, type_of_file)
+    name.downcase! if name =~ /[A-Z]/
+    # some filenames had spaces instead of underscores
+    name.tr!(" ", "_") if name.include?(" ")
+    # some filenames have more than one underscore but we want to match
+    #  the style of existing sound filenames with one underscore
+    name.gsub!(/_{2,}/, "_") if name =~ /_{2,}/
+    name.gsub!(/^\_+/, '') if name =~ /^\_+/
+
+    if name.chomp([-4..-1].to_s) =~ /\_+$/
+      name.gsub!(/\_+$/, '')
+      name.chomp!([-4..-1].to_s)
+      name << ".mp3" if type_of_file == "csv"
+    end
+
+    return name
+  end
+
+  def convert_wav_to_mp3(folder_path)
+    # converts .wav files to .mp3
+    puts %x(
+        for file in "#{folder_path}"/*.wav
+        do \`ffmpeg -i "$file" "${file%.wav}".mp3\`
+          done
+      )
+    Dir.mkdir "#{folder_path}/mp3_files"
+
+    Dir[folder_path + "/*.mp3"].each do |f|
+      filename = File.basename(f, File.extname(f))
+
+      if filename =~ /(^\_+|\_+$|_{2,}|\s|[A-Z])/
+        fixed_name = fix_corrupted_filename(filename, "audio")
+        File.rename(f, folder_path + "/" + fixed_name + File.extname(f))
+        f.gsub!(f.split('/')[-1].to_s, "#{fixed_name}.mp3")
+      end
+
+      FileUtils.mv(f.to_s, "#{folder_path}/mp3_files")
+    end
+  end
+
+  def parse_filenames_in_csv(csv)
+    filename = csv.by_col[1]
+    # changed file extension to .mp3 to run .generateSoundMetaData.rb
+    return filename.each do |name|
+      # original file extensions are .wav but they shoud be .mp3
+      # in order to run ./generateSoundMetaData.rb
+      name.gsub!(".wav", ".mp3")
+
+      if name =~ /(^\_+|\_+$|_{2,}|\s|[A-Z])/
+        fix_corrupted_filename(name, "csv")
+      end
+    end
+  end
+
+  def parse_aliases_in_csv(csv)
+    # removed spaces to simplify
+    # because csv file had aliases that had
+    # more than one space or none
+    aliases_with_spaces_removed = csv.by_col[3].each do |aliase|
+      aliase.gsub!(/,\s+/, ',').downcase!
+      aliase.tr!(" ", ',')
+    end
+
+    return aliases_with_spaces_removed.map do |aliase|
+      "\"#{aliase.split(/\,/).join('\', \'').gsub!(/,\s+/, ',')}\""
+    end
+  end
+
+  def create_json
+    filename = parse_filenames_in_csv(@csv)
+    parsed_aliases = parse_aliases_in_csv(@csv)
+
+    sound_hash = {}
+    filename.zip(parsed_aliases).each do |name, aliases|
+      sound_hash[name.to_s] = aliases
+    end
+
+    folder_path = ARGV[0]
+    convert_wav_to_mp3(folder_path)
+
+    sound_hash.each do |name, aliases|
+      Dir[folder_path + "/mp3_files/*"].each do |file_path|
+        regex = /^#{name}$/
+
+        if file_path.split('/')[-1].downcase =~ /(^\_+|\_+$|_{2,}|\s|[A-Z])/
```

Comment:
This regex shows up several times in your code - maybe worth extracting to a constant.

Reply:
Of course! Should have thought of that! I'll make it a constant.